test/e2e_node/device_plugin.go

```diff
@@ -176,20 +211,13 @@ func makeBusyboxPod(resourceName, cmd string) *v1.Pod {
 	}
 }
 
-// parseLogFromNRuns returns restart count of the specified container
-// after it has been restarted at least restartCount times,
-// and the matching string for the specified regular expression parsed from the container logs.
-func parseLogFromNRuns(f *framework.Framework, podName string, contName string, restartCount int32, re string) (int32, string) {
-	var count int32
+// parseLogFromNRuns returns the matching string for the specified regular expression parsed from the container logs.
+func parseLogFromNRuns(f *framework.Framework, podName string, contName string, re string) string {
```

Comment:
If this function don't return the restart count, suggest to rename it to `parseLog()`.

Reply:
I think the reason this function returns restartCount is to make sure pod gets restarted after the last failure event, and the returned log is from the recent run. Otherwise, we run with the risk that the returned log is still from last run before the failure event and so is not really testing the functionality. Is it possible that we include both GetPodLogs and RestartCount checking in the Eventually loop?

Reply:
@jiayingz current logic is not very intuitive and readable.
> to make sure pod gets restarted after the last failure event, and the returned log is from the recent run.

I think you meant container restarts because pod never restart. In that case how about adding a clear specific logic to get a pod container restart and after ensuring that container has got restarted,. this function is used to read logs to verify that device assignment still intact?